File: Juan/P1/Ruby/Napakalaki/pkg/Napakalaki-0.0.1/lib/Monster.rb
#encoding: utf-8
#Versión 1.0
# author: juane, Raúl

require_relative 'BadConsequence.rb'
require_relative 'TreasureKind.rb'
require_relative  'Prize.rb'

class Monster
    #Atributos propios de la clase
  
    # @name # String para representar el nombre del monstruo
    # @combatLevel # int para representar el nivel de combate del monstruo
  
    #Otros atributos
    # @badconsequence  # de tipo BadConsequence
    # @price # de tipo Prize
 
  
  
    def initialize(name,level,bc,price)
        @name = name
        @combatLevel = level    
        @price = price
        @badconsequence = bc
    
    end
  
     attr_accessor :name
     attr_accessor :combatLevel
     attr_accessor :price
     attr_accessor :badconsequence
    def to_s
        "Monster =  #{@name}  , combatLevel =   #{@combatLevel} ,  #{@badconsequence} ,  Price:  #{@price} "      
    end
  
end

File: Raul/P2/Java/Napakalaki/src/napakalaki/Monster.java
package napakalaki;

/**
 *
 * @author juane y Raúl
 * @version 1.1
 * Clase Monster
 */
public class Monster {
      
    /* Atributos propios de la clase*/
    
    private String name; // nombre del monstruo
    private int combatLevel; // nivel de combate del monstruo

    /* Atributo de tipo Prize y BadConsequence */
    private Prize price;
    private BadConsequence badConsequence;
    
    /* Constructor*/
    
    Monster(String name, int level, BadConsequence bc, Prize price){
        
        this.name = name;
        this.combatLevel = level;
        this.badConsequence = bc;
        this.price = price;
    }
    
    /**
     * Método getName()
     * Devuelve el nombre del monstruo
     * @return name de tipo int
     */
    public String getName() {
        return name;
    }

    /**
     * Método getCombatLevel()
     * Devuelve el nivel de combate del monstruo
     * @return 
     */
    public int getCombatLevel() {
        return combatLevel;
    }

    /**
     * Método getPrice()
     * Devuelve el buen royo
     * @return  price de tipo Prize
     */
    public Prize getPrice() {
        return price;
    }

    /**
     * Método getBadConsequence()
     * Devuelve el mal royo
     * @return badConsequence de tipo BadConsequence
     */
    public BadConsequence getBadConsequence() {
        return badConsequence;
    }

    
    
    
    
    /**
     * Método toString
     * @return 
     */
    @Override
    public String toString() {
        return "Monster = " + this.name + ", combatLevel = " 
                +  this.combatLevel + ", Price: " +  this.price 
                + ", " + this.badConsequence ;
    }
}
